version: "3.7"

services:

  nginx:
    image: nginx:1.19.5
    ports:
      - 80:80
    networks:
      - gis_6ban_network
    volumes:
      - /home/ubuntu/nginx.conf:/etc/nginx/nginx.conf
      - gis_6ban_static:/data/static/
      - gis_6ban_media:/data/media/

  django:
    image: gis_6ban_django:4
    networks:
      - gis_6ban_network
    volumes:
      - gis_6ban_static:/home/gis_6ban_1/staticfiles/
      - gis_6ban_media:/home/gis_6ban_1/media/
    secrets:
      - DJANGO_SECRET_KEY

  mariadb:
    image: mariadb:10.5
    networks:
      - gis_6ban_network
    volumes:
      - gis_mariadb:/var/lib/mysql
    environment:
      MARIADB_ROOT_PASSWORD: password1234
      MARIADB_DATABASE: django
      MARIADB_USER: django
      MARIADB_PASSWORD: password1234


networks:
  gis_6ban_network:

volumes:
  gis_6ban_static:
  gis_6ban_media:
  gis_mariadb:

secrets:
  DJANGO_SECRET_KEY:
    external: true
